tcl/is.tcl

```diff
@@ -534,10 +266,58 @@ namespace eval qc::is {
         return [expr {[string toupper $bool] in {Y N YES NO TRUE FALSE T F 0 1}}]
     }
 
-    proc decimal {number} {
+    proc decimal {args} {
         #| Checks if the given number is a decimal.
+        qc::args $args -precision ? -scale ? -- number
         if {[string is double -strict $number]} {
```

Comment:
PostgreSQL numeric/decimals:
> up to 131072 digits before the decimal point; up to 16383 digits after the decimal point

So they can get very large. Perhaps better to check the count of digits before and after a decimal point rather than checking if it's a double.